packages/trader/src/AppV2/Components/TradeParameters/Duration/duration.tsx

```diff
@@ -12,24 +12,47 @@ type TDurationProps = {
 };
 
 const Duration = observer(({ is_minimized }: TDurationProps) => {
-    const { duration, duration_unit, expiry_time, expiry_type } = useTraderStore();
+    const { duration, duration_unit, expiry_time, expiry_type, contract_type, onChange } = useTraderStore();
     const { name_plural, name } = getUnitMap()[duration_unit] ?? {};
     const duration_unit_text = name_plural ?? name;
     const [selected_hour, setSelectedHour] = useState<number[]>([]);
     const [is_open, setOpen] = useState(false);
+    const [unit, setUnit] = useState(expiry_time ? 'et' : duration_unit);
+
+    const handleHour = React.useCallback(() => {
+        if (expiry_time) {
+            setUnit('et');
+        } else {
+            // eslint-disable-next-line no-lonely-if
+            if (duration_unit === 'm' && duration > 59) {
+                const hour = Math.floor(duration / 60);
+                const minutes = duration % 60;
+                setUnit('h');
+                setSelectedHour([hour, minutes]);
+            } else {
+                setSelectedHour([]);
+                setUnit(duration_unit);
+            }
+        }
+    }, [duration, duration_unit, expiry_time]);
 
     useEffect(() => {
-        if (duration_unit === 'm' && duration > 59) {
-            const hour = Math.floor(duration / 60);
-            const minutes = duration % 60;
-            setSelectedHour([hour, minutes]);
+        handleHour();
+        if (
+            ['turboslong', 'turbosshort', 'high_low', 'touch'].includes(contract_type) &&
```

Comment:
If you want, this strings can be replaced with TRADE_TYPES from (packages/shared/src/utils/contract/contract.tsx).
Just for my understanding, what is so special about these contracts, that we need to hardcode some check?

Reply:
Actually these are the types on which ticks can be selected form  5 to 10 . lemme check if we have anything in the store to get it dynamically